src/main/java/org/broadinstitute/hellbender/tools/walkers/qc/CheckPileup.java

```diff
@@ -0,0 +1,164 @@
+import java.util.List;
+
+/**
+ * Compare GATK's internal pileup to a reference Samtools pileup
+ *
+ * <p>
+ *     At every locus in the input set, compares the mpileup data (reference base, aligned base from
+ *     each overlapping read, and quality score) generated internally by GATK to a reference pileup data generated
+ *     by Samtools. Only consider single-sample mpileup format.
+ *     See the <a href="http://samtools.sourceforge.net/pileup.shtml">Pileup format documentation</a> for more details
+ *     on the format
+ * </p>
+ *
+ * <h3>Input</h3>
+ * <p>
+ *     A BAM file containing your aligned sequence data and a mpileup file generated by Samtools covering the region you
+ *     want to examine.
+ * </p>
+ *
+ * <h3>Output</h3>
+ * <p>
+ *     A text file listing mismatches between the input mpileup and the GATK's internal pileup. If there are no mismatches,
+ *     the output file is empty.
+ * </p>
+ *
+ * <h3>Usage example</h3>
+ * <pre>
+ * ./gatk-launch CheckPileup \
+ *   -R reference.fasta \
+ *   -I your_data.bam \
+ *   --pileup pileup_file.pileup \
+ *   -L chr1:257-275 \
+ *   -O output_file_name
+ * </pre>
+ */
+@CommandLineProgramProperties(
+    summary = "At every locus in the input set, compares the pileup data (reference base, aligned base from each overlapping"
+        + "read, and quality score) generated internally by GATK to a reference pileup data generated by Samtools.",
+    oneLineSummary = "Compare GATK's internal pileup to a reference Samtools mpileup",
+    programGroup = QCProgramGroup.class)
+public final class CheckPileup extends LocusWalker {
+
+    /**
+     * This is the existing mpileup against which we'll compare GATK's internal pileup at each genome position in the desired interval.
+     */
+    @Argument(fullName = "pileup", shortName = "pileup", doc="Pileup generated by Samtools")
+    public FeatureInput<SAMPileupFeature> mpileup;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "Output file (if not provided, defaults to STDOUT).", optional = true)
+    public File outFile = null;
+
+    /**
+     * By default the program will quit if it encounters an error (such as missing truth data for a given position).
+     * Use this flag to override the default behavior; the program will then simply print an error message and move on
+     * to the next position.
+     */
+    @Argument(fullName="continue_after_error", doc="Continue after encountering an error", optional=true)
+    public boolean continueAfterAnError = false;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    private long nLoci = 0;
+    private long nBases = 0;
+    private PrintStream out;
+
+    @Override
+    public void onTraversalStart() {
+        try {
+            out = (outFile == null) ? System.out : new PrintStream(outFile);
+        } catch (FileNotFoundException e) {
```

Comment:
Sorry for this extra boiler plate.  We should improve the output stream handling in the future  part of #121 )